pdk/1.0/modules/ROOT/pages/policies-pdk-architecture.adoc

```diff
@@ -0,0 +1,42 @@
+= Policy Development Kit Architecture Overview
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:imagesdir: ../assets/images
+
+The Policy Development Kit (PDK) consists of the:
+
+* <<anypoint-cli-pdk-plugin>>: Generates and distributes the policy to Exchange
+* <<policy-template>>: Scaffolds new custom policies
+* <<sdk-building-tools, SDK Building Tools>>: Simplifies the development process
+
+== Anypoint CLI PDK Plugin
+
+The Anypoint CLI PDK plugin creates the PDK project and uploads policy assets to Exchange. After creating the project, PDK provides a simplied set of commands in a `Makefile` to complete all following development steps.
+
+== Policy Template
+
+PDK generates some simple scaffolding when creating the policy.
+
+All generated files contain the required information to properly compile the policy.
+
+For more information about the included files, see xref:policies-pdk-create-project.adoc#project-structure[Project Structure].
+
+== SDK Building Tools
+
+Flex Gateway is built on Envoy which requires developed policies to be compatible with the event-driven https://github.com/proxy-wasm/spec[proxy-wasm^] architecture. 
+
+`proxy-wasm`'s event-driven architecture forces users to keep track of all the different events received to handle a single request. This causes the following drawbacks that make developing and maintaining event driven code costly:
+
+* High complexity of the code due to the coupling of business logic with the entry points 
+* Inability to modularize common logic which creates monolithic code
+* Verbose non-linear control flow that create difficult function callbacks and error propagation
+
+PDK provides building tools that abstract the user from the event oriented coding approach by using reactor and executor patterns. PDK provides the developer a way to code each incoming request in a linear method which results in: 
```

Comment:
```suggestion
PDK provides building tools that abstract the developer from the event-oriented coding approach by using reactor and executor patterns. PDK provides the developer a way to code each incoming request in a linear method which results in: 
```
I assume "user" is the same as "developer"? I was a little confused; I wonder if the customer will also be confused?